hack/generated/pkg/genruntime/base_types.go

```diff
@@ -17,6 +20,14 @@ type MetaObject interface {
 	KubernetesResource
 }
 
+type Reconciler interface { // TODO: Sorta awkward interface name
+	CreateOrUpdate(ctx context.Context) (ctrl.Result, error)
+
+	Delete(ctx context.Context) (ctrl.Result, error)
+
+	// TODO: Do we want a "Diff"? Or is the expectation that logic just exists in CreateOrUpdate?
```

Comment:
I don't think so - at least, not until we have a proven need for it. Keeping the interface as minimal as possible is useful.